File: src/Klc.LogicRoute.Domain/Enums/ShipmentStatus.cs
namespace Klc.LogicRoute.Domain.Enums;

public enum ShipmentStatus
{
    Draft = 0,
    Calculated = 1,
    PendingApproval = 2,
    Approved = 3,
    SentToProvider = 4,
    VehicleAssigned = 5,
    Loading = 6,
    InTransit = 7,
    Delivered = 8,
    Completed = 9,
    Cancelled = 10
}

File: mobile/composeApp/src/commonMain/kotlin/com/klc/logicroute/data/model/ShipmentStatus.kt
package com.klc.logicroute.data.model

import kotlinx.serialization.Serializable

@Serializable
enum class ShipmentStatus {
    Assigned,
    Loaded,
    InTransit,
    Delivered,
    Cancelled;

    fun displayName(): String = when (this) {
        Assigned -> "Atandı"
        Loaded -> "Yüklendi"
        InTransit -> "Yolda"
        Delivered -> "Teslim Edildi"
        Cancelled -> "İptal"
    }
}
